### PR TITLE
adding buffering to the plugin

### PR DESCRIPTION
**Pull request by nick1907**
Hi all,
I'd like to present you the feature which I've added to this plugin, I've added ability to set the **size of buffer** instead of using default **4KB**. I've tested this changes on my production environment and it can make the publishing faster even 10 times.  To do that I needed to add a new version of **comons-io** (2.5). I tried to do as little changes in the project structure as it possible.  I hope you will like the my "feature".

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -116,8 +122,8 @@
             <artifactId>easymockclassextension</artifactId>
             <version>2.4</version>
             <scope>test</scope>
-        </dependency>
-
```
Comment (slide) [inline pom.xml]: whitespace only changes need to be fixed up

File: pom.xml [hunk 2]
```diff
@@ -142,8 +148,9 @@
                 <artifactId>maven-hpi-plugin</artifactId>
 				<!--  <version>1.67</version> -->
                 <extensions>true</extensions>
-                <configuration>
-                    <compatibleSinceVersion>0.2</compatibleSinceVersion>
+		<configuration>
+			<pluginFirstClassLoader>true</pluginFirstClassLoader>
+			<!-- <compatibleSinceVersion>0.2</compatibleSinceVersion> -->
```
Comment (slide) [inline pom.xml]: Why was this commented out?

File: src/main/java/jenkins/plugins/publish_over_cifs/CifsClient.java [hunk 3]
```diff
@@ -106,7 +108,7 @@ public void transferFile(final CifsTransfer transfer, final FilePath filePath, f
         if (buildInfo.isVerbose()) buildInfo.println(Messages.console_copy(helper.hideUserInfo(newFileUrl)));
         final OutputStream out = createFile(newFileUrl).getOutputStream();
         try {
-            IOUtils.copy(content, out);
+            IOUtils.copy(content, out, this.bufferSize);
```
Comment (slide) [inline src/main/java/jenkins/plugins/publish_over_cifs/CifsClient.java]: No need to specify "this"

File: src/main/java/jenkins/plugins/publish_over_cifs/CifsHostConfiguration.java [hunk 4]
```diff
@@ -95,7 +101,7 @@ public CifsClient createClient(final BPBuildInfo buildInfo) {
         final String url = buildUrl(false);
         final NtlmPasswordAuthentication auth = new NtlmPasswordAuthentication(getDomain(), getUsername(false), getPassword()); 
         testConfig(url, auth);
-        return new CifsClient(buildInfo, url, auth);
+        return new CifsClient(buildInfo, url, auth, this.bufferSize);
```
Comment (slide) [inline src/main/java/jenkins/plugins/publish_over_cifs/CifsHostConfiguration.java]: No need to specify "this."

File: src/main/java/jenkins/plugins/publish_over_cifs/descriptor/CifsPublisherPluginDescriptor.java [hunk 5]
```diff
@@ -49,6 +49,7 @@
 import org.kohsuke.stapler.QueryParameter;
 import org.kohsuke.stapler.StaplerRequest;
 import org.kohsuke.stapler.StaplerResponse;
+import sun.misc.FormattedFloatingDecimal;
```
Comment (slide) [inline src/main/java/jenkins/plugins/publish_over_cifs/descriptor/CifsPublisherPluginDescriptor.java]: It doesn't look like this import is used.

File: pom.xml (hunk 2 above)
Comment (nick1907) [reply pom.xml]: Sorry, I forgot to uncoment it. I'll fix this.

File: pom.xml (hunk 1 above)
Comment (nick1907) [reply pom.xml]: My text editor has some problems with spaces. Sorry for that.

**Comment by nick1907** [comment]
My newest change (333c0513ac02e784c3f10c4e3d1adf4ed294d513) has fixed the problems with spaces etc.

File: src/main/resources/jenkins/plugins/publish_over_cifs/Messages.properties [hunk 6]
```diff
@@ -37,6 +37,7 @@ console.config.noWins=Removing WINS from name resolution
 console.config.wins=Setting WINS server [{0}]
 console.config.timout=Setting response timeout [{0}]
 console.config.soTimeout=Setting socket timeout [{0}]
+console.config.bufferSize=Setting buffer size [{0}]
```
Comment (slide) [inline src/main/resources/jenkins/plugins/publish_over_cifs/Messages.properties]: I don't see this actually being used in the code, is it used somewhere?

Comment (nick1907) [reply src/main/resources/jenkins/plugins/publish_over_cifs/Messages.properties]: I completely forgot to use it, thank you for that information. Btw. I've added the usage of that command in the newest change => f2bcff9b4c045b9f9cfd0c29a3f6ddbe955078cb

**Pull request merged**